drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -609,10 +609,46 @@ DEVICE_DT_INST_DEFINE(inst,                                        \
 
 DT_INST_FOREACH_STATUS_OKAY(GPIO_ITE_DEV_CFG_DATA)
 
+/* To prevent cc pins leakage, disables integrated cc module. */
+static void it8xxx2_disable_cc_module(int port)
+{
+	/* Power down all CC, and disable CC voltage detector */
+	IT8XXX2_USBPD_CCGCR(port) |= USBPD_REG_MASK_DISABLE_CC;
+	IT8XXX2_USBPD_CCGCR(port) |= USBPD_REG_MASK_DISABLE_CC_VOL_DETECTOR;
+	/*
+	 * Disconnect CC analog module (ex.UP/RD/DET/TX/RX), and
+	 * disconnect CC 5.1K to GND
+	 */
+	IT8XXX2_USBPD_CCCSR(port) |= (USBPD_REG_MASK_CC2_DISCONNECT |
+				      USBPD_REG_MASK_CC2_DISCONNECT_5_1K_TO_GND |
+				      USBPD_REG_MASK_CC1_DISCONNECT |
+				      USBPD_REG_MASK_CC1_DISCONNECT_5_1K_TO_GND);
+	/* Disconnect CC 5V tolerant */
+	IT8XXX2_USBPD_CCPSR(port) |= (USBPD_REG_MASK_DISCONNECT_POWER_CC2 |
+				      USBPD_REG_MASK_DISCONNECT_POWER_CC1);
+}
```

Comment:
I'm wondering - shouldn't the USBPD parts be separated into a separate driver, which in turn uses the GPIO subsystem itself. Adding these changes as a part of the preexisting GPIO driver doesn't necessarily sound like a good idea.

Reply:
So far ITE EC it8XXX2 series (ex.it81202, it81302) support two   USBPD ports on chip, and the notebook platform may use 0/1/2 ITE on chip USBPD port. We need to take care all the cases, even if the platform doesn't use ITE USBPD port, then it won't enable ITE USBPD driver and we still need to disable cc pins to prevent leakage (then pins can be used as
gpio: GPH1, GPH2, GPF4, GPF5),  so we implemented this part in GPIO subsystem.